### PR TITLE
nsc-events-nextjs_7_383_roleauthorized-my-events-page

### LINKED ISSUE #383: Role-based authorization: My Event page
Tasks:

- [x] Restrict access to My Event page
- [x] Redirect the user to page with 401 Unauthorized status code and "You are not authorized to access this page" message when attempting to access these pages

### PR DESCRIPTION
**Pull request by kaimanasse15**
Resolves #383
Only the admin can get into this page to see the content

### CODE REVIEW

**Review by heosman — APPROVED** [review]
LGTM! There are some branch conflicts you need to resolve, though.
image

**Comment by cshimm** [comment]
Looks good to me I saw the Unauthorized page when I was logged out and the My Events page when I was logged in. The conflicts don't look too complex but let me know if you need help resolving them!

**Review by JesseCaddell — APPROVED** [review]
Looks good to me. Works as intended. As the others have mentioned just some conflicts that need to be resolved!
image

**Comment by kaimanasse15** [comment]
> Looks good to me I saw the Unauthorized page when I was logged out and the My Events page when I was logged in. The conflicts don't look too complex but let me know if you need help resolving them!

Hi @cshimm, I am wordering what is the conflict? So many projects are going for me that's why I am not sure which one. Thank you.

**Review by Ali-7s — APPROVED** [review]
LGTM.

image


There seems to be a simple ESLint error though, create-events page is missing a bracket at the end.

image

File: app/create-event/page.tsx [hunk 1]
```diff
@@ -75,16 +78,7 @@ const CreateEvent: React.FC = () => {
...
-        <Box sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '80vh' }}>
-          <Typography variant="h5" component="h2">
-            You are not authorized to access this page.
-          </Typography>
-        </Box>
-      );
-    }
-  
-  
+
```
Comment (IsaacJrTypes) [inline app/create-event/page.tsx:81]: Hi Kai, can you explain your reasoning for leaving this condition statement empty?

File: components/UnauthorizedPages.tsx
Comment (IsaacJrTypes) [inline components/UnauthorizedPages.tsx:1]: Nice job on isolating the box component into a separate component.  I would rename the component to UnauthorizedPageBox or UnauthorizedPageMessage, so it's clear what's in the component.

**Review by IsaacJrTypes — COMMENTED** [review]
Double check your merge commit. I am getting errors in create-event page
CleanShot 2024-05-30 at 22 26 52@2x

**Review by heosman — CHANGES_REQUESTED** [review]
Like others have mentioned, the condition statement is empty and the `UnauthorizedPages` component is unused on the Create Event page so I am still able to access the page as a non-admin and non-creator user. There is also a missing curcly brace and semicolon before the export statement at the end of the Create Event file. 

I think it would be great to implement @IsaacJrTypes's suggestion for clarity purposes.

**Comment by kaimanasse15** [comment]
@heosman I tried to fix it again. Please help me to double check on it. Thank you

**Comment by bennettsf** [comment]
I feel like it would be nice if you were consistent with how you implement the role checking. 

Maybe do the same thing you did in 'my-events' for 'create-event' where it's just **one return with an if else statement** rather than an isolated if statement:

uOstDLY (1)


Other than that and following Isaac's suggestion to rename the component, everything looks good!

**Comment by cshimm** [comment]
> > Looks good to me I saw the Unauthorized page when I was logged out and the My Events page when I was logged in. The conflicts don't look too complex but let me know if you need help resolving them!
> 
> Hi @cshimm, I am wordering what is the conflict? So many projects are going for me that's why I am not sure which one. Thank you.

if you go to the bottom of the page you should see a message notifying you about conflicts with 'main'
image

if you click that you'll be brought to a basic text editor that shows the code for the file with conflict and 2 blocks of code wrapped with a red line. 
image

These blocks of code are in different branches and are conflicting with each other, so they'll need to be merged manually. You'll need to go through both blocks carefully, including your changes from your branch and avoiding overwriting or deleting any code from 'main'.

**Comment by heosman** [comment]
> @heosman I tried to fix it again. Please help me to double check on it. Thank you

@kaimanasse15 I've tested your PR again and there are still some things you need to fix. 
I am experiencing a compile error because of the missing curly brace before the export statement. Here is how you would fix that:
image

The conditions in the if-else statements on the My Events page and Create Event are different, but that shouldn't be the case. Because of the incorrect `if` condition and the lack of the `else` statement, which should be returning the Unauthorized page, the Create Event page is still displaying for unauthorized users. Like @bennettsf mentioned, being consistent with the conditional logic and re-using the Unauthorizedpages component the same way you did on the My Event page should simplify things.  You should also check out @cshimm's instructions of how to resolve the branch conflicts.

**Comment by kaimanasse15** [comment]
> > @heosman I tried to fix it again. Please help me to double check on it. Thank you
> 
> @kaimanasse15 I've tested your PR again and there are still some things you need to fix. I am experiencing a compile error because of the missing curly brace before the export statement. Here is how you would fix that: image
> 
> The conditions in the if-else statements on the My Events page and Create Event are different, but that shouldn't be the case. Because of the incorrect `if` condition and the lack of the `else` statement, which should be returning the Unauthorized page, the Create Event page is still displaying for unauthorized users. Like @bennettsf mentioned, being consistent with the conditional logic and re-using the Unauthorizedpages component the same way you did on the My Event page should simplify things. You should also check out @cshimm's instructions of how to resolve the branch conflicts.

Yeah it showed error when I add one more in. May you passed the full code for me on Slack? Thanks, @heosman

**Comment by heosman** [comment]
>
> 
> Yeah it showed error when I add one more in. May you passed the full code for me on Slack? Thanks, @heosman
I sent you the code with the curly brace added in on Slack.

**Comment by taylorpapke** [comment]
> @kaimanasse15 I sent you the code with the curly brace added in on Slack.

Hey @heosman, there's been no activity on this PR since you commented a week ago. Do you mind pushing the fix you have to this PR so we can get it merged in?

**Review by heosman — APPROVED** [review]

**Pull request merged**